### PR TITLE
[TS SDK] Support `transfer` fungible token

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
This PR adds 
1. support `transfer` a fungible token.
2. `getTokenData` to support token v2 standard
### Test Plan
tests are passing

### CODE REVIEW

File: ecosystem/typescript/sdk/src/plugins/aptos_token.ts [hunk 1]
```diff
@@ -484,7 +501,42 @@ export class AptosToken {
...
+      const fungibleAsset = new FungibleAssetClient(this.provider);
+      const txnHash = await fungibleAsset.transfer(
+        data.owner,
+        data.tokenAddress,
+        data.recipient,
+        data.amount,
+        data.extraArgs,
+      );
+      return txnHash;
+    }
```
[the remark this answers is not in the document]
Comment (0xmaayan) [reply ecosystem/typescript/sdk/src/plugins/aptos_token.ts]: agreed. Updated the function to accept an optional `isFungibleToken`, if it is not set, the function would query indexer to check if the token type

File: ecosystem/typescript/sdk/src/plugins/aptos_token.ts [hunk 2]
```diff
@@ -484,7 +501,50 @@ export class AptosToken {
...
+   * for the token data and check whether it is a non-fungible or a fungible token.
+   *
+   * @param data NonFungibleTokenParameters | FungibleTokenParameters type
+   * @param isFungibleToken (optional) The token type, non-fungible or fungible token.
+   * @returns The hash of the transaction submitted to the API
+   */
+  async transfer(
+    data: NonFungibleTokenParameters | FungibleTokenParameters,
+    isFungibleToken?: boolean | null,
+  ): Promise<string> {
```
Comment (bowenyang007) [inline ecosystem/typescript/sdk/src/plugins/aptos_token.ts:522]: Does this work for token v1?

Comment (0xmaayan) [reply ecosystem/typescript/sdk/src/plugins/aptos_token.ts:522]: no, `aptos_token` is for token v2. `token_client` is for token v1.

Comment (gregnazario) [reply ecosystem/typescript/sdk/src/plugins/aptos_token.ts:522]: I think this is a good callout for token v1 confusion, but I think we should be moving towards the ideal experience for token V2

**Review by gregnazario — APPROVED** [review]

**Comment by 0xmaayan** [comment]
> I think this is a good callout for token v1 confusion, but I think we should be moving towards the ideal experience for token V2

Yes, tokenv1 and tokenv2 api interfaces are pretty different, so making one generic function for each operation that can support both v1 and v2 is too much. Since we anyway want to move toward the v2 world, itmakes sense to have a v2 aptos token client.

File: ecosystem/typescript/sdk/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,9 @@ All notable changes to the Aptos Node SDK will be captured in this file. This ch
 - export `bcsSerializeU256` from `bcs/helper.ts`
 - Add `examples` folder to `.npmignore`
 - use `0x1::aptos_account::transfer` in tests
+- Support transfer a fungible token.
+  - add a `transfer` function to the `AptosToken` class that accepts `NonFungibleTokenParameters` or `FungibleTokenParameters` types.
```
Comment (banool) [inline ecosystem/typescript/sdk/CHANGELOG.md:11]: Nit:
```suggestion
  - Add a `transfer` function to the `AptosToken` class that accepts `NonFungibleTokenParameters` or `FungibleTokenParameters` types.
```

File: ecosystem/typescript/sdk/src/plugins/aptos_token.ts [hunk 4]
```diff
@@ -484,7 +501,50 @@ export class AptosToken {
       [HexString.ensure(token).hex(), HexString.ensure(recipient).hex()],
     );
     const bcsTxn = AptosClient.generateBCSTransaction(owner, rawTxn);
-    const pendingTransaction = await this.provider.aptosClient.submitSignedBCSTransaction(bcsTxn);
+    const pendingTransaction = await this.provider.submitSignedBCSTransaction(bcsTxn);
     return pendingTransaction.hash;
   }
+
+  /**
+   * Transfer a token. This function supports transfer non-fungible token and fungible token.
```
Comment (banool) [inline ecosystem/typescript/sdk/src/plugins/aptos_token.ts:509]: Could we add a comment saying this will not work for token v1?

**Review by banool — APPROVED** [review]

**Pull request merged**